### PR TITLE
[SPARK-36547][BUILD] Downgrade scala-maven-plugin to 4.3.0

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

When preparing Spark 3.2.0 RC1, I hit the same issue of https://github.com/apache/spark/pull/31031. 
```
[INFO] Compiling 21 Scala sources and 3 Java sources to /opt/spark-rm/output/spark-3.1.0-bin-hadoop2.7/resource-managers/yarn/target/scala-2.12/test-classes ...
[ERROR] ## Exception when compiling 24 sources to /opt/spark-rm/output/spark-3.1.0-bin-hadoop2.7/resource-managers/yarn/target/scala-2.12/test-classes
java.lang.SecurityException: class "javax.servlet.SessionCookieConfig"'s signer information does not match signer information of other classes in the same package
java.lang.ClassLoader.checkCerts(ClassLoader.java:891)
java.lang.ClassLoader.preDefineClass(ClassLoader.java:661)
```
This PR is to apply the same fix again by downgrading scala-maven-plugin to 4.3.0

### Why are the changes needed?

To unblock the release process.

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

Build test

### CODE REVIEW

**Comment by gengliangwang** [comment]
I am still trying the fix with ./dev/create-release/do-release-docker.sh. It will probably work just like https://github.com/apache/spark/pull/31031.

**Comment by cloud-fan** [comment]
@HyukjinKwon @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Oh. Thank you for pinging me, @cloud-fan and @gengliangwang .

File: pom.xml [hunk 1]
```diff
@@ -2574,7 +2574,8 @@
         <plugin>
           <groupId>net.alchim31.maven</groupId>
           <artifactId>scala-maven-plugin</artifactId>
-          <version>4.5.3</version>
+          <!-- SPARK-36547: Please don't upgrade the version below, otherwise there will be an error on building Hadoop 2.7 package -->
+          <version>4.3.0</version>
```
Comment (dongjoon-hyun) [inline pom.xml:2578]: My bad. I forgot the history. :(

**Review by dongjoon-hyun — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (HyukjinKwon) [reply pom.xml:2578]: No worries .. It was only able to reproduce in some specific environments. I don't think there was anything wrong when we upgrade.

Comment (HyukjinKwon) [reply pom.xml:2578]: In fact, ideally it should be investigated why it fails in the specific environments .. but I am fine to unblock the release for now.

**Comment by gengliangwang** [comment]
Merging to master/3.2

**Pull request closed without merging**